File: java-core/src/main/java/org/maw/wedding/navigation/NavigationEventController.java
package org.maw.wedding.navigation;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class NavigationEventController implements EventListener {

    private Navigator mNavigator;
    private Bus mEventBus;
    private Map<Integer, ScreenEvent> mNavigationMap = new HashMap<>();

    public NavigationEventController(Navigator navigator, Bus eventBus) {
        mNavigator = navigator;
        mEventBus = eventBus;
    }

    public void registerForScreenEvents(List<ScreenEvent> eventList) {
        for (ScreenEvent screenEvent : eventList) {
            mNavigationMap.put(screenEvent.getId(), screenEvent);
        }
    }

    public void onReady() {
        mEventBus.subscibe(Bus.EVENT_TYPE.NAVIGATION, this);
    }

    public void cleanUp() {
        mEventBus.unSubscribe(Bus.EVENT_TYPE.NAVIGATION, this);
    }

    @Override
    public void onEvent(Event event) {
        if (mNavigationMap.containsKey(event.getEventId())) {
            navigate(mNavigationMap.get(event.getEventId()));
        }
    }

    private void navigate(ScreenEvent screenEvent) {
        mNavigator.navigate(screenEvent);
    }
}

File: kotlin-core/src/main/kotlin/org/maw/wedding/navigation/NavigationEventController.kt
package org.maw.wedding.navigation

import java.util.*


class NavigationEventController(val mNavigator: Navigator, val mEventBus: Bus) : EventListener {
    private val mNavigationMap = HashMap<Int, ScreenEvent>()

    fun registerForScreenEvents(eventList: List<ScreenEvent>) {
        for (screenEvent in eventList) {
            mNavigationMap.put(screenEvent.id, screenEvent)
        }
    }

    fun onReady() {
        mEventBus.subscribe(this)
    }

    fun cleanUp() {
        mEventBus.unSubscribe(this)
    }

    override fun onEvent(event: Event) {
        if (mNavigationMap.containsKey(event.id)) {
            navigate(mNavigationMap[event.id]!!)
        }
    }

    private fun navigate(screenEvent: ScreenEvent) {
        mNavigator.navigate(screenEvent)
    }
}
